xbmc/settings/DisplaySettings.cpp

```diff
@@ -656,10 +656,39 @@ RESOLUTION CDisplaySettings::GetResolutionForScreen()
   return RES_DESKTOP;
 }
 
+static inline bool ModeSort(std::pair<std::string, std::string> i,std::pair<std::string, std::string> j)
+{
+  return (i.second > j.second);
+}
+
+void CDisplaySettings::SettingOptionsModesFiller(std::shared_ptr<const CSetting> setting, std::vector< std::pair<std::string, std::string> > &list, std::string &current, void *data)
+{
+  RESOLUTION res = CDisplaySettings::GetInstance().GetDisplayResolution();
+  RESOLUTION_INFO info = CDisplaySettings::GetInstance().GetResolutionInfo(res);
+
+  for (auto index = (unsigned int)RES_DESKTOP; index < CDisplaySettings::GetInstance().ResolutionInfoSize(); ++index)
```

Comment:
modes start at RES_CUSTOM

Reply:
I had it that way initially but had complaints that two of the same resolutions would be shown

Reply:
I had it that way initially but had complaints that two of the same resolutions would be shown

Reply:
so you removed the double entry by increasing the list? magic :)

Reply:
I misremembered, it was that some destoop resolution werensh't own in the whitelist. But we've figured out why on slack 😝